src/hardware/pros_controller.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <memory>
+
+#include "interface/controller.h"
+#include "pros.h"
+
+namespace hardware {
+pros::controller_analog_e_t ConvertProsJoystick(
+    interface::ControllerJoystick joystick);
+
+class ProsController : public interface::Controller {
+ public:
+  ProsController(pros::controller_id_e_t id) : controller_(id) {}
+
+  static std::unique_ptr<ProsController> MakeDriverController();
+  static std::unique_ptr<ProsController> MakeOperatorController();
```

Comment:
I don't think I'm a fan of `Driver` and `Operator` for the two controller designations (since there is no restriction on which controller does what). It might be better here to reference `Primary` and `Secondary`, as in the order that communication happens with the controllers directly

Reply:
I chose `Driver` and `Operator` because that's usually what it is in FRC, although I with the inherent limitations of such phrasing. I like the use of `Primary` and `Secondary` better though, yeah